src/bjointsp/template/component.py

```diff
@@ -19,11 +19,14 @@ def __init__(self, name, type, stateful, inputs, outputs, cpu, mem, dr, config=N
         self.outputs_back = outputs[1]
         self.cpu = cpu      # function of forward and backward ingoing data rates
         self.mem = mem
+        self.vnf_delay = vnf_delay
         self.dr = dr[0]
         self.dr_back = dr[1]
         self.config = config		# config used by external apps/MANOs (describes image, ports, ...)
 
         total_inputs = self.inputs + self.inputs_back
+
+        # What is this used for? For every input we need a total of 1 CPU and 1 Mem? 
```

Comment:
As discussed last time, the number of CPU/memory coefficients depends on the number of inputs. If we have a VNF with 2 inputs, we'll need 3 CPU parameters: a*lambda1+b*lambda2+c

This simply checks that the number of CPU parameters matches the number of inputs. It does not mean that we have to have a total of 1 CPU or 1 Memory.